lib/lookup/src/lookup_v2/owned.rs

```diff
@@ -96,6 +99,34 @@ impl Serialize for OwnedPath {
     }
 }
 
+fn serialize_field(field: &str, separator: Option<&str>) -> String {
+    let needs_quotes = field
+        .chars()
+        .any(|c| !matches!(c, 'A'..='Z' | 'a'..='z' | '_' | '0'..='9' | '@'));
```

Comment:
Might be good to add a comment here, pointing to the VRL parser, to maker sure we keep the list of allowed characters in sync.